libs/wire-api/src/Wire/API/Team/Member.hs

```diff
@@ -258,11 +259,18 @@ modelNewTeamMember = Doc.defineModel "NewTeamMember" $ do
     Doc.description "the team member to add"
 
 instance ToJSON NewTeamMember where
-  toJSON t = object ["member" .= _ntmNewTeamMember t]
+  toJSON t = object ["member" .= mem]
+    where
+      mem = Object . HM.fromList . fltr . HM.toList $ o
+      Object o = toJSON . _ntmNewTeamMember $ t
+      fltr = filter ((`elem` ["user", "permissions", "created_by", "created_at"]) . fst)
 
 instance FromJSON NewTeamMember where
-  parseJSON = withObject "add team member" $ \o ->
-    NewTeamMember <$> o .: "member"
+  parseJSON = withObject "add team member" $ \o -> do
+    mem <- o .: "member"
+    if (_legalHoldStatus mem == defUserLegalHoldStatus)
```

Comment:
I can't find where `defUserLegalHoldStatus` is defined.

In general I wonder: wouldn't it be cleaner to change NewTeamMember to take a different type to `TeamMember` (a new one without legalhold_status), which would then make adding if statements and filtering not necessary. Should be more readable also.
(In case you'd like to stick to filtering and conditionals - could you add some comments explaining why that's done for the the future reader?)